### DISCUSSION TITLE
Breakout Cablee

### DISCUSSION THREAD

**Discussion by remchuk** [netbox-community/netbox#14408] [Help Wanted!]
Hey there!

So I was trying to create Breakout Cables in the Netbox, but it seems quite Impossible.

The only 2 bandage solutions are either creating virtual subinterfaces, which is bad because there is no way to create a cable to the subinterface.

The other solution is creating Physical interface, which is also not right because there is no really new interfaces but rather one interface split.

Is there a best practice? Or is there a planned best practice?

**Comment by candlerb** [comment]
See #11917, #12369

**Reply by remchuk** [reply]
Meaning that there is no real solution right now. Just a bandage solution.

But the question is, Is there something in the Plans?

**Comment by jeremystretch** [comment]
> The other solution is creating Physical interface, which is also not right because there is no really new interfaces but rather one interface split.

If you're utilizing channelized interfaces (e.g. 4x10GE), this is the correct way to model these interfaces. Each channel is a discrete physical termination.

**Reply by Flabenelli** [reply]
> > The other solution is creating Physical interface, which is also not right because there is no really new interfaces but rather one interface split.
> 
> If you're utilizing channelized interfaces (e.g. 4x10GE), this is the correct way to model these interfaces. Each channel is a discrete physical termination.

So how you document that p.e swp30 has a breakout cable connected? and the relatioship between the swp30 and the others terminations? swp30s[0-3]

**Reply by candlerb** [reply]
> So how you document that p.e swp30 has a breakout cable connected?

You create a cable with one end in swp30, and the other end with 4 terminations in whatever the target interfaces are.

(Aside: you can't have a mixture of frontports and interfaces. You can only terminate into 4 interfaces *or* 4 frontports))

> and the relatioship between the swp30 and the others terminations? swp30s[0-3]

That currently isn't modelled.

You can create four virtual interfaces swp30s[0-3], and you can set the "parent" interface of all four to be swp30.  However, you cannot trace from swp30s0 (say) and find the correct single remote interface. That's because:

1. in a cable with multiple terminations, the terminations are not ordered;
2. when there are multiple virtual interfaces with the same parent, those aren't ordered either;
3. you currently can't trace from a virtual interface (even if its parent is physical).

If you want that level of tracing then you'll have to create swp30s[0-3] as 4 physical interfaces, and run *separate* cables into each one (i.e. use 4 individual cables to represent your breakout cable)
